File: dotnet/01-commands/tests/MultiplesOf3or5/MultiplesOf3or5TestTemplate1.cs
namespace MultiplesOf3or5
{
    public class MultiplesOf3or5TestTemplate1
    {
        [Fact]
        public void GivenNegativeInput_Return0()
        {
            var multiplesOf3or5 = new MultiplesOf3or5(-1);
            Assert.Equal(0, multiplesOf3or5.Sum());
        }

        [Fact]
        public void Given0_Return0()
        {
            var multiplesOf3or5 = new MultiplesOf3or5(0);
            Assert.Equal(0, multiplesOf3or5.Sum());
        }

        [Fact]
        public void Given4_Return3()
        {
            var multiplesOf3or5 = new MultiplesOf3or5(4);
            Assert.Equal(3, multiplesOf3or5.Sum());
        }

        [Fact]
        public void Given10_Return23()
        {
            var multiplesOf3or5 = new MultiplesOf3or5(10);
            Assert.Equal(23, multiplesOf3or5.Sum());
        }

        [Fact]
        public void Given20_Return78()
        {
            var multiplesOf3or5 = new MultiplesOf3or5(20);
            Assert.Equal(78, multiplesOf3or5.Sum());
        }

        [Fact]
        public void Given200_Return9168()
        {
            var multiplesOf3or5 = new MultiplesOf3or5(200);
            Assert.Equal(9168, multiplesOf3or5.Sum());
        }
    }
}

File: java/01-commands/tests/sourcecode/src/test/java/com/training/app/MultiplesOf3or5/MultiplesOf3or5TestTemplate1.java
package com.training.app.MultiplesOf3or5;

import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.assertEquals;

public class MultiplesOf3or5TestTemplate1 {

    @Test
    public void givenNegativeInput_return0() {
        MultiplesOf3or5 multiplesOf3or5 = new MultiplesOf3or5(-1);
        assertEquals(0, multiplesOf3or5.sum());
    }

    @Test
    public void given0_return0() {
        MultiplesOf3or5 multiplesOf3or5 = new MultiplesOf3or5(0);
        assertEquals(0, multiplesOf3or5.sum());
    }

    @Test
    public void given4_return3() {
        MultiplesOf3or5 multiplesOf3or5 = new MultiplesOf3or5(4);
        assertEquals(3, multiplesOf3or5.sum());
    }

    @Test
    public void given10_return23() {
        MultiplesOf3or5 multiplesOf3or5 = new MultiplesOf3or5(10);
        assertEquals(23, multiplesOf3or5.sum());
    }

    @Test
    public void given20_return78() {
        MultiplesOf3or5 multiplesOf3or5 = new MultiplesOf3or5(20);
        assertEquals(78, multiplesOf3or5.sum());
    }

    @Test
    public void given200_return9168() {
        MultiplesOf3or5 multiplesOf3or5 = new MultiplesOf3or5(200);
        assertEquals(9168, multiplesOf3or5.sum());
    }
}
